### PR TITLE
Create `BottomUpReconciliator`

### LINKED ISSUE #1037: Create `BottomUpReconciler`
### 🚀 Feature Request

Create reconciler implementing Bottom-Up approach

### Proposal

1. Create class `BottomUpReconciler(BaseReconciler)`
- Check that `source_level` is lower than `target_level`
3. Implement method `fit`:
    - Receive dataset on the the level which is lower or equal than `target_level`, `source_level`
    - Aggregate the dataset to the `source_level`
    - Set mapping matrix as summing matrix from source to target level 

### Test cases
- Test constructor works correctly with different source and target levels
- Test method fit saves the correct matrix in `mapping_matrix` for different source/target levels
    - source=target
    - source<target
    - source>target

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by brsnw250**
## Before submitting (must do checklist)

- [ ] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [ ] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [ ] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #1037

### CODE REVIEW

File: etna/reconciliation/bottom_up.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        target_level:
+            Level to be reconciled from the forecasts.
+        source_level:
+            Level to be forecasted.
+        """
+        super().__init__(target_level=target_level, source_level=source_level)
+
+    def fit(self, ts: TSDataset) -> "BottomUpReconciliator":
+        """
+        Fit the reconciliator parameters.
```
Comment (alex-hse-repository) [inline etna/reconciliation/bottom_up.py:22]: Start the description on the same line with quotes

File: etna/reconciliation/bottom_up.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        source_level_index = ts.hierarchical_structure.get_level_depth(self.source_level)
+        target_level_index = ts.hierarchical_structure.get_level_depth(self.target_level)
+
+        if source_level_index < target_level_index:
+            raise ValueError("Source level should be lower or equal in the hierarchy than the target level!")
+
+        if current_level_index < source_level_index:
+            raise ValueError("Current TSDataset level should be lower or equal in the hierarchy than the source level!")
+
+        if (ts[..., "target"] < 0).values.any():
```
Comment (alex-hse-repository) [inline etna/reconciliation/bottom_up.py:47]: This method works correctly with the values of any kind

**Review by alex-hse-repository — CHANGES_REQUESTED** [review]

File: etna/reconciliation/bottom_up.py (hunk 2 above)
Comment (brsnw250) [reply etna/reconciliation/bottom_up.py:47]: Added this check for consistency purpose only. I think in this case we should add info about nonnegative numbers to `TopDownReconciliator` doc.

Comment (alex-hse-repository) [reply etna/reconciliation/bottom_up.py]: Good idea, may you do it in this PR

**Review by alex-hse-repository — APPROVED** [review]

**Pull request merged**